src/compas/files/gltf.py

```diff
@@ -0,0 +1,629 @@
+from __future__ import print_function
+from __future__ import absolute_import
+
+
+__all__ = [
+    'GLTF',
+    'GLTFReader',
+    'GLTFParser',
+]
+
+import base64
+import io
+import json
+import os
+import struct
+from math import fsum
+
+from compas.geometry import identity_matrix
+from compas.geometry import matrix_from_quaternion
+from compas.geometry import matrix_from_scale_factors
+from compas.geometry import matrix_from_translation
+from compas.geometry import multiply_matrices
+from compas.geometry import transpose_matrix
+
+COMPONENT_TYPE_ENUM = {
+    5120: 'b',  # BYTE
+    5121: 'B',  # UNSIGNED_BYTE
+    5122: 'h',  # SHORT
+    5123: 'H',  # UNSIGNED_SHORT
+    5125: 'I',  # UNSIGNED_INT
+    5126: 'f',  # FLOAT
+}
+
+NUM_COMPONENTS_BY_TYPE_ENUM = {
+    "SCALAR": 1,
+    "VEC2": 2,
+    "VEC3": 3,
+    "VEC4": 4,
+    "MAT2": 4,
+    "MAT3": 9,
+    "MAT4": 16,
+}
+
+DEFAULT_ROOT_NAME = 'root'
+
+
+class MeshData(object):
+    """Object containing COMPAS consumable data for creating a mesh.
+
+    Attributes
+    ----------
+    vertices : list
+        List of the xyz-coordinates of the vertices of the mesh.
+    faces : list of tuples
+        List of the faces of the mesh represented by tuples of vertices referenced by index.
+    mesh_name : str
+        Name of the mesh, if any.
+    """
+    def __init__(self, faces, vertices, mesh_name):
+        self.vertices = vertices
+        self.faces = faces
+        self.mesh_name = mesh_name
+        # self.material = ??
+
+
+class GLTFScene(object):
+    """Object representing a root node of a scene.
+
+    Attributes
+    ----------
+    name : str
+        Name of the scene, if any.
+    edges : list
+        List of 2-tuples referencing nodes by key.
+    nodes : dict
+        Dictionary of (node_key, GLTFNode) pairs.
```

Comment:
```suggestion
        Dictionary of (node_key, :class:`GLTFNode`) pairs.
```
Here we can use the class directive thingy to have these linked/clickable on the doc.